src/index.test.ts

```diff
@@ -18,13 +18,17 @@ describe('createStreamMiddleware', () => {
     const initRes = { id: 1, jsonrpc };
     const res = { id: 1, jsonrpc, result: 'test' };
 
-    await new Promise<void>((resolve, reject) => {
-      // listen for incoming requests
-      jsonRpcConnection.stream.on('data', (_req) => {
-        expect(req).toStrictEqual(_req);
-        jsonRpcConnection.stream.write(res);
-      });
+    // listen for incoming requests
+    jsonRpcConnection.stream.on('data', (_req) => {
+      expect(req).toStrictEqual(_req);
+      jsonRpcConnection.stream.write(res);
+    });
 
+    // wait for the stream to be ready
+    await artificialDelay();
```

Comment:
Huh, interesting, I didn't realize the behavior of the stream was different after waiting for the next event loop. I can see that this is needed for the test to catch the broken case though.